### ISSUE TITLE
strict-comparison false positive (TSLint 5.18.0)

### ISSUE THREAD

**Issue by vladeck** [palantir/tslint#4774]
### Bug Report

- __TSLint version__: 5.18.0
- __TypeScript version__: 3.5.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const foo: string | null = null;

if (foo !== null) { // <-- Cannot use '!==' comparator for type 'null'. (strict-comparisons)tslint(1)
    // ...
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:all",
    "rules": {
        "linebreak-style": false,
        "no-null-keyword": false,
        "strict-type-predicates": false
    }
}
```

#### Actual behavior

`strict-comparison` error.

#### Expected behavior

No error.

**Comment by ulrichb** [comment]
Same here. The exclusion for "if one of the operands is null or undefined" doesn't seem to work.

**Comment by ulrichb** [comment]
... and in addition to the sample above it should also work for non-primitives on the other side right?

Example:

```ts
const someObj : {} | undefined = {};
if (someObj === undefined) { /* ... */ }
```
